gps_common/src/utm_odometry_node.cpp

```diff
@@ -11,166 +11,194 @@
+      odom_tf.transform.rotation.z = odom.pose.pose.orientation.z;
+      odom_tf.transform.rotation.w = odom.pose.pose.orientation.w;
+
+      odom_tf_broadcaster.sendTransform(odom_tf);
+    }
+
+    void INSPVACallBack(const novatel_gps_msgs::InspvaConstPtr& ins) {
+
+      double northing, easting, z;
+      std::string zone;
+      LLtoUTM(ins->latitude, ins->longitude, northing, easting, zone);
+      if (ignore_z)
+        z = 0.0;
+      else
+        z = ins->height;
+
+      Eigen::Vector3d position(easting, northing, z);
+      Eigen::Matrix3d rot_mat = world_to_local_frame.second.toRotationMatrix();
+      Eigen::Vector3d local_position = rot_mat * position + world_to_local_frame.first;
+      if (ignore_z) 
+        local_position = Eigen::Vector3d(local_position(0), local_position(1), 0.0);
+
+      Eigen::Quaterniond orientation = Eigen::AngleAxisd(ins->roll * DEG2RAD, Eigen::Vector3d::UnitX()) *
+        Eigen::AngleAxisd(ins->pitch * DEG2RAD, Eigen::Vector3d::UnitY()) *
+        Eigen::AngleAxisd((90.0 - ins->azimuth) * DEG2RAD, Eigen::Vector3d::UnitZ());
+      Eigen::Quaterniond local_orientation = world_to_local_frame.second * orientation;
+
+      odom.header.stamp = ins->header.stamp;
+      odom.pose.pose.position.x = local_position(0);
+      odom.pose.pose.position.y = local_position(1);
+      odom.pose.pose.position.z = local_position(2);
+      odom.pose.pose.orientation.w = local_orientation.w();
+      odom.pose.pose.orientation.x = local_orientation.x();
+      odom.pose.pose.orientation.y = local_orientation.y();
+      odom.pose.pose.orientation.z = local_orientation.z();
+
+      double velocity = sqrt(pow(ins->east_velocity,2) + pow(ins->north_velocity,2) + pow(ins->up_velocity,2));
+      odom.twist.twist.linear.x = velocity;
+      odom.twist.twist.linear.y = 0; //TODO:
+      odom.twist.twist.linear.z = 0; //TODO:
+
+      // TODO use covariance from inscov
+      odom.pose.covariance = {1,0,0,0,0,0,
+                0,1,0,0,0,0,
+                0,0,1,0,0,0,
+                0,0,0,1,0,0,
+                0,0,0,0,1,0,
+                0,0,0,0,0,1};
+      odom_pub.publish(odom);
+
+      if (publish_odom_tf)
+        PublishOdomTF(odom);
+
+      // keep twist and covariance matrix
+      odom_world.pose = odom.pose;
+      odom_world.header.stamp = odom.header.stamp;
+      // replace timestamp, position and orientation
+      odom_world.pose.pose.position.x = position(0);
+      odom_world.pose.pose.position.y = position(1);
+      odom_world.pose.pose.position.z = position(2);
+      odom_world.pose.pose.orientation.w = orientation.w();
+      odom_world.pose.pose.orientation.x = orientation.x();
+      odom_world.pose.pose.orientation.y = orientation.y();
+      odom_world.pose.pose.orientation.z = orientation.z();
+      world_odom_pub.publish(odom_world);
+    }
+
+  public:
+    UTMOdometryNode() : priv_node("~") {}
+
+    void Run() {
```

Comment:
member functions are camelCase according to google style guide.

- [ ] change to "run()"